From: Ignacio Losiggio <iglosiggio@gmail.com>
Subject: Fixed various segfaults caused by bad casts

* src/forcepad/MainFrame2.cpp
* src/forcepad/MainFrame2_kiosk.cpp
* src/forcepad/TabletToolbar.cpp
* src/forcepad/TabletToolbar_kiosk.cpp

Fixed using the following sed command:
  sed -i "s/->setView(/->setView((CPaintView*) /" \
    src/forcepad/MainFrame2.cpp \
    src/forcepad/MainFrame2_kiosk.cpp \
    src/forcepad/TabletToolbar.cpp \
    src/forcepad/TabletToolbar_kiosk.cpp 

--- a/src/forcepad/MainFrame2.cpp
+++ b/src/forcepad/MainFrame2.cpp
--- MainFrame2.cpp.orig	2017-06-08 15:12:49.880629966 -0300
+++ MainFrame2.cpp	2017-06-08 15:13:31.612105330 -0300
@@ -141,7 +141,7 @@
 
 void CMainFrame::cb_settingsMenuGeneral_i(Fl_Menu_*, void*) {
   CGeneralSettings* generalSettings = new CGeneralSettings();
-generalSettings->setView(paintView);
+generalSettings->setView((CPaintView*) paintView);
 generalSettings->centerWindow(wndMain);
 generalSettings->show();
 delete generalSettings;
@@ -2092,7 +2092,7 @@
 static Fl_RGB_Image image_structure_black_grd(idata_structure_black_grd, 36, 36, 3, 0);
 
 void CMainFrame::cb_btnVizMixer_i(Fl_HoverButton*, void*) {
-  m_vizMixer->setView(paintView);
+  m_vizMixer->setView((CPaintView*) paintView);
 m_vizMixer->show();
 m_vizMixer->placeWindow(wndMain);
 }
@@ -11063,7 +11063,7 @@
   m_tabletToolbar->setView(paintView);
   
   m_vizMixer = new CVizMixerToolbar();
-  m_vizMixer->setView(paintView);
+  m_vizMixer->setView((CPaintView*) paintView);
   
   m_optSettings = new COptSettings();
   m_optSettings->setView(paintView);
--- MainFrame2_kiosk.cpp.orig	2017-06-08 15:13:07.512408304 -0300
+++ MainFrame2_kiosk.cpp	2017-06-08 15:13:31.636105028 -0300
@@ -1801,7 +1801,7 @@
 static Fl_RGB_Image image_rectangle_black(idata_rectangle_black, 36, 36, 4, 0);
 
 void CMainFrame::cb_btnVizMixer_i(Fl_HoverButton*, void*) {
-  m_vizMixer->setView(paintView);
+  m_vizMixer->setView((CPaintView*) paintView);
 m_vizMixer->show();
 m_vizMixer->placeWindow(wndMain);
 }
@ -10840,7 +10843,7 @@
   m_tabletToolbar->setView(paintView);
   
   m_vizMixer = new CVizMixerToolbar();
-  m_vizMixer->setView(paintView);
+  m_vizMixer->setView((CPaintView*) paintView);
   
   m_optSettings = new COptSettings();
   m_optSettings->setView(paintView);
--- TabletToolbar_kiosk.cpp.orig	2017-06-08 15:13:22.568219027 -0300
+++ TabletToolbar_kiosk.cpp	2017-06-08 15:13:31.640104978 -0300
@@ -636,7 +636,7 @@
 	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	
 	CCalcSettings* calcSettings = new CCalcSettings();
-	calcSettings->setView(view);
+	calcSettings->setView((CPaintView*) view);
 	calcSettings->centerWindow(mainWindow);
 	calcSettings->show();
 	delete calcSettings;
--- TabletToolbar.cpp.orig	2017-06-08 15:13:27.792153353 -0300
+++ TabletToolbar.cpp	2017-06-08 15:13:31.636105028 -0300
@@ -15,7 +15,7 @@
 void CTabletToolbar::cb_btnOpen_i(Fl_HoverButton*, void*) {
   if (m_paintView!=NULL)
 {
-	CPaintView* view = (CPaintView*)m_paintView;
+	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	view->openModel();
 };
 }
@@ -324,7 +324,7 @@
 void CTabletToolbar::cb_btnNew_i(Fl_HoverButton*, void*) {
   if (m_paintView!=NULL)
 {
-	CPaintView* view = (CPaintView*)m_paintView;
+	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	view->newModel();
 };
 }
@@ -632,7 +632,7 @@
 void CTabletToolbar::cb_btnSaveAs_i(Fl_HoverButton*, void*) {
   if (m_paintView!=NULL)
 {
-	CPaintView* view = (CPaintView*)m_paintView;
+	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	view->saveModelAs();
 };
 }
@@ -940,7 +940,7 @@
 void CTabletToolbar::cb_btnUndo_i(Fl_HoverButton*, void*) {
   if (m_paintView!=NULL)
 {
-	CPaintView* view = (CPaintView*)m_paintView;
+	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	view->undo();
 };
 }
@@ -1248,7 +1248,7 @@
 void CTabletToolbar::cb_btnSettings_i(Fl_HoverButton*, void*) {
   if (m_paintView!=NULL)
 {
-	CPaintView* view = (CPaintView*)m_paintView;
+	CFlPaintView* view = (CFlPaintView*)m_paintView;
 	
 	CCalcSettings* calcSettings = new CCalcSettings();
 	calcSettings->setView(view);
